platform/x11/os_x11.cpp

```diff
@@ -240,28 +242,14 @@ Error OS_X11::initialize(const VideoMode &p_desired, int p_video_driver, int p_a
 
 // maybe contextgl wants to be in charge of creating the window
 #if defined(OPENGL_ENABLED)
-	// Set DRI_PRIME if not set. This means that Godot should default to a higher-power GPU if it exists.
-	// Note: Due to the final '0' parameter to setenv any existing DRI_PRIME environment variables will not
-	// be overwritten.
-	bool enable_dri_prime = true;
-	// Check if Nouveau is loaded, we don't want to force dGPU usage with that driver.
-	if (FileAccess *f = FileAccess::open("/proc/modules", FileAccess::READ)) {
-		// Match driver name + space
-		String nouveau_str = "nouveau ";
-
-		while (!f->eof_reached()) {
-			String line = f->get_line();
-
-			if (line.begins_with(nouveau_str)) {
-				enable_dri_prime = false;
-				break;
-			}
+	if (getenv("DRI_PRIME") == NULL) {
+		print_line("DRI_PRIME not set. Attempting to detect best GPU.");
```

Comment:
I'd make this line `print_verbose` to avoid spamming people who don't have a dGPU setup needlessly.
Then the other `print_line` could maybe also get another `print_verbose` with more details, e.g.:
```
print_line("Found discrete GPU, setting DRI_PRIME=1 to use it.");
print_verbose("Set DRI_PRIME in the environment to override the GPU detection logic.");
```

Reply:
Having information on what GPUs were identified could be interesting too as `print_verbose` in `detect_prime()` maybe?